### ISSUE TITLE
User object not modifiable through Sentry singleton

### ISSUE THREAD

**Issue by mgray88** [getsentry/sentry-java#917]
The User class itself is mutable, however there is no `getUser` to go with the `setUser` in the Sentry class. It seems I am able to modify it through the `configureScope` function. Is this the only way to do it, or are there plans to expose it through the Sentry singleton?

**Comment by tlf30** [comment]
It would be great to get access to the user object. There are situations where the user id is known independent of the username and being able to update the user object as data becomes known would be very helpful instead of recreating the object every time.

**Comment by bruno-garcia** [comment]
There are currently no plans to expose a getter via `Sentry` static class.

Some background why we have `Sentry.setUser`:

The original design was to mutate scope data through `configureScope`, and now it's odd that we have two ways of doing things. This creeped in because of folks using raven-js upgrading to sentry-javascript.

In JS they also have only set and no get. In fact the scope in many SDKs have no getter at all.

I agree this API isn't ideal. IMHO we should drop this static getter from the `Sentry` class and reduce the API surface there.

**Comment by tlf30** [comment]
@bruno-garcia Is there any means of accessing the user object at this time?

**Comment by marandaneto** [comment]
@tlf30 https://github.com/getsentry/sentry-java/issues/917#issuecomment-622793697

tldr: use the `configureScope` method.

I'll close this as won't fix as we offer this feature over `configureScope`

**Comment by bruno-garcia** [comment]
As @marandaneto said, to sumarize:

```java
Sentry.configureScope { 
  // Modify the user in the scope
  User user = scope.getUser();
} 
```
